### PR TITLE
SCP-2206: Add a warning card for when cached wallet details are not found.

### PR DESCRIPTION
**Pull request by merivale**
We are currently showing the pickup wallet card on page load if the user has a previously picked up wallet in their `localStorage`. But if the deployment has been updated, all data (including previously used wallets) will have disappeared. This PR changes it so that we first check whether the wallet still exists on the server - if it doesn't, we show a warning to the user inviting them to clear their browser's cache (because "cache" is likely a more familiar word - in fact, we clear the `localStorage`).

localhost_8009_ (5)

### CODE REVIEW

File: marlowe-dashboard-client/src/Pickup/State.purs [hunk 1]
```diff
@@ -99,3 +102,10 @@ handleAction PickupWallet = do
   walletLibrary <- use _walletLibrary
   liftEffect $ setItem walletLibraryLocalStorageKey $ encodeJSON walletLibrary
   callMainFrameAction $ MainFrame.EnterPlayState walletLibrary walletDetails
+
+handleAction ClearLocalStorage = do
+  liftEffect $ removeItem walletLibraryLocalStorageKey
+  liftEffect $ removeItem walletDetailsLocalStorageKey
+  window_ <- liftEffect window
+  location_ <- liftEffect $ location window_
```
Comment (hrajchert) [inline marlowe-dashboard-client/src/Pickup/State.purs:110]: very minor, but 

```purescript
location_ <- liftEffect <<< location <<< window
```

**Review by hrajchert — APPROVED** [review]
LGTM

File: marlowe-dashboard-client/src/Pickup/State.purs (hunk 1 above)
Comment (merivale) [reply marlowe-dashboard-client/src/Pickup/State.purs:110]: Compiler says no. :confused: I feel there must be a way to simplify this, but for some reason I can't see what it is at the moment.

Comment (hrajchert) [reply marlowe-dashboard-client/src/Pickup/State.purs:110]: sorry, my bad :P 

```purescript
handleAction ClearLocalStorage = do
  liftEffect do
    removeItem walletLibraryLocalStorageKey
    removeItem walletDetailsLocalStorageKey
  location_ <- liftEffect $ location =<< window
  liftEffect $ reload location_
```

**Pull request merged**